deepchecks/core/check_json.py

```diff
@@ -0,0 +1,114 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing the check results classes."""
+# pylint: disable=super-init-not-called
+import io
+from typing import Dict, List, Union
+
+import jsonpickle
+import pandas as pd
+import plotly
+
+from deepchecks.core.check_result import CheckFailure, CheckResult
+from deepchecks.core.condition import (Condition, ConditionCategory,
+                                       ConditionResult)
+
+__all__ = [
+    'CheckResultJson',
+    'CheckFailureJson',
+]
+
+
+class FakeCheck:
+    def __init__(self, metadata: Dict):
+        self._metadata = metadata
+
+    def metadata(self, *args, **kwargs):  # pylint: disable=unused-argument
+        return self._metadata
+
+    def name(self):
+        return self._metadata['name']
+
+
+class CheckResultJson(CheckResult):
+    """Class which returns from a check with result that can later be used for automatic pipelines and display value.
+
+    Class containing the result of a check
+
+    The class stores the results and display of the check. Evaluating the result in an IPython console / notebook
+    will show the result display output.
+
+    Parameters
+    ----------
+    json_data: Union[str, Dict]
+        Json data
+    """
+
+    def __init__(self, json_dict: Union[str, Dict]):
+        if isinstance(json_dict, str):
+            json_dict = jsonpickle.loads(json_dict)
+
+        self.value = json_dict.get('value')
+        self.header = json_dict.get('header')
+        self.check = FakeCheck(json_dict.get('check'))
```

Comment:
why we need this `FakeCheck` type?
could not we recreate an original check?
we store a check type name and its parameters, I think that it is enough info to recreate an original check type instance/

Reply:
we can't create a check class from that and we can't use those functions